src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -199,7 +204,7 @@ protected internal bool CheckWitness(byte[] hashOrPubkey)
             {
                 20 => new UInt160(hashOrPubkey),
                 33 => Contract.CreateSignatureRedeemScript(ECPoint.DecodePoint(hashOrPubkey, ECCurve.Secp256r1)).ToScriptHash(),
-                _ => throw new ArgumentException(null, nameof(hashOrPubkey))
+                _ => throw new CatchableException("The argument should be a script hash or public key.")
```

Comment:
I don't think we should be throwing catchable exception in CheckWitness for security reasons